engine/api/notification/user.go

```diff
@@ -198,61 +198,67 @@ type UserNotificationInput struct {
+		JOIN 	pipeline ON pipeline.id = application_pipeline.pipeline_id
+		JOIN 	environment ON environment.id = environment_id
+		JOIN 	project ON project.id = pipeline.project_id
+		WHERE 	project.projectkey = $1
+		ORDER BY pipeline.name`
+		args = []interface{}{projectKey}
+	} else {
+		query = `
+		SELECT 	application_pipeline_id, environment_id, settings, pipeline.id, pipeline.name, environment.name
+		FROM  	application_pipeline_notif
+		JOIN 	application_pipeline ON application_pipeline.id = application_pipeline_notif.application_pipeline_id
+		JOIN 	pipeline ON pipeline.id = application_pipeline.pipeline_id
+		JOIN 	environment ON environment.id = environment_id
+		JOIN 	project ON project.id = pipeline.project_id
+		JOIN	application_group ON application_pipeline.application_id = application_group.application_id
+		JOIN 	pipeline_group ON pipeline.id = pipeline_group.pipeline_id
+		JOIN 	group_user ON group_user.group_id = pipeline_group.group_id AND group_user.group_id = application_group.group_id
+		WHERE 	project.projectkey = $1
+		AND 	group_user.user_id = $2
+		ORDER BY pipeline.name`
+		args = []interface{}{projectKey, u.ID}
 	}
 
-	var errParse error
-	notif1.Notifications, errParse = ParseUserNotificationSettings(settingsBody)
-	return notif1, errParse
-}
-
-//ParseUserNotificationSettings transforms json to UserNotificationSettings map
-func ParseUserNotificationSettings(settings []byte) (map[sdk.UserNotificationSettingsType]sdk.UserNotificationSettings, error) {
-	mapSettings := map[string]interface{}{}
-	if err := json.Unmarshal(settings, &mapSettings); err != nil {
+	rows, err := db.Query(query, args...)
+	if err != nil {
 		return nil, err
 	}
+	defer rows.Close()
+	for rows.Next() {
+		var un sdk.UserNotification
+		var settings string
+		err := rows.Scan(&un.ApplicationPipelineID, &un.Environment.ID, &settings, &un.Pipeline.ID, &un.Pipeline.Name, &un.Environment.Name)
+		if err != nil {
+			return nil, err
+		}
+		un.Notifications, err = sdk.ParseUserNotificationSettings([]byte(settings))
+		if err != nil {
+			return nil, err
+		}
 
-	notifications := map[sdk.UserNotificationSettingsType]sdk.UserNotificationSettings{}
-
-	for k, v := range mapSettings {
-		switch k {
-		case string(sdk.EmailUserNotification), string(sdk.JabberUserNotification):
-			if v != nil {
-				var x sdk.JabberEmailUserNotificationSettings
-				tmp, err := json.Marshal(v)
-				if err != nil {
-					log.Warning("ParseUserNotificationSettings> unable to parse JabberEmailUserNotificationSettings: %s", err)
-					return nil, sdk.ErrParseUserNotification
-				}
-				if err := json.Unmarshal(tmp, &x); err != nil {
-					log.Warning("ParseUserNotificationSettings> unable to parse JabberEmailUserNotificationSettings: %s", err)
-					return nil, sdk.ErrParseUserNotification
-				}
-				notifications[sdk.UserNotificationSettingsType(k)] = &x
+		if u != nil {
+			if !permission.AccessToEnvironment(un.Environment.ID, u, permission.PermissionRead) {
```

Comment:
Must check if environment is not default environment or modify AccessToEnvironment to return truee if envID = 1